### DISCUSSION TITLE
Is there a way to have ContactListener generate contact information for kinematic vs static interactions?

### DISCUSSION THREAD

**Discussion by LeifNo** [jrouwe/JoltPhysics#1236] [Q&A] [answered]
I understand that kinematic and static bodies both have infinite mass and it's therefore the user's responsibility to perform any additional logic to handle separating them, but is there a way to have ContactListener still return the contact points when they come into contact? I.e., for OnContactAdded to give a result where one body is Kinematic and one is Static, even with the knowledge that the constraint system will not solve this collision on its own.

If not, is the proper way to query this type of information to do a shape cast?

Thank you.

**Comment by mihe** [comment, marked as answer]
See `BodyCreationSettings::mCollideKinematicVsNonDynamic` and `Body::SetCollideKinematicVsNonDynamic`:

https://github.com/jrouwe/JoltPhysics/blob/8feb90cf41af1aef61e0d51085b61d882a4597d7/Jolt/Physics/Body/Body.h#L76-L79

**Reply by LeifNo** [reply]
Oh interesting, thank you. I had seen mention of that property on this page https://jrouwe.github.io/JoltPhysics/index.html, but it was under the Sensors section so I must have not put together that it wasn't ONLY applicable to sensors.
